### PR TITLE
[SYCL] Fix pi2ur conversion of bool info queries

### PR DESCRIPTION
**Pull request by callumfare**
Also fix:
* Uninitialized stype values in UR queries
* Missing PI_KERNEL_INFO_NUM_REGS mapping

### CODE REVIEW

File: sycl/plugins/unified_runtime/pi2ur.hpp [hunk 1]
```diff
@@ -410,14 +410,16 @@ inline pi_result ur2piUSMAllocInfoValue(ur_usm_alloc_info_t ParamName,
 }
 
 // Handle mismatched PI and UR type return sizes for info queries
-inline pi_result fixupInfoValueTypes(size_t ParamValueSizeUR,
+inline pi_result fixupInfoValueTypes(size_t ParamValueSizeRetUR,
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/pi2ur.hpp:413]: I think we need to drop this function entirely and pull in this change: https://github.com/oneapi-src/unified-runtime/pull/556

Comment (kbenzie) [reply sycl/plugins/unified_runtime/pi2ur.hpp:413]: We just discussed this in the UR WG and had agreed not to change UR and to separate out these changes from the CUDA port.

Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/pi2ur.hpp:413]: What is the motivation to not define boolean in UR as a 32-bit integer?
It already causes issues her, and in https://github.com/intel/llvm/pull/9654 and in https://github.com/intel/llvm/pull/9561

Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/pi2ur.hpp:413]: > Overall I don't beleive changing the size of ur_bool_t is the correct way to address a bug in the pi2ur translation layer.

If we didn't have the mismatch between PI and UR in a boolean type size, the there would be no "mismatch" that this `fixupInfoValueTypes` is trying to address (with a bug in it). I still think making `ur_bool_t` be 32-bit integer has more pros than cons (no cons actually). The plugins always assumed boolean values are 32-bit integers so they are more robust this way.

Comment (kbenzie) [reply sycl/plugins/unified_runtime/pi2ur.hpp:413]: I had another look at #9561, it changed from writing a 32-bit to 8-bit value to the out parameter, this is actually doing the opposite of what you're asking for and is inconsistent with the change added to the L0 port in #9654.

For the sake of argument, if we accepted the 32-bit `ur_bool_t` change, the change to using the 8-bit bool for those queries would reintroduce an issue to `sycl-ls` by only writing the lower 8-bits and leaving the upper 24-bits as garbage.

As for existing plugins assuming the boolean type is 32-bit, that's not actually true for the ports the my team has is working on (CUDA, HIP, and OpenCL). We have been striving to implement the UR spec as it is written today, with its 8-bit `ur_bool_t` definition.

Like I said before, its unfortunate that `fixupInfoValueTypes` had a bug but I still do not believe forcing through a change to `ur_bool_t` being 32-bit is the correct way to fix this.

Stepping back from the idea that _it's always been 32-bit so it should stay that way_ I think the fact that #9561 is using the 8-bit `bool` type shows that users do not expect boolean types to be 32-bits.

Level Zero also defines its boolean `ze_bool_t` as `uint8_t` [here](https://github.com/oneapi-src/level-zero/blob/master/include/ze_api.h#L95).

Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/pi2ur.hpp:413]: Of course, technically boolean can be represented by 8-bits . My argument was that making it 32-bit in UR would get in sync with most other existing uses, and not require conversions. I don't feel strongly about 8 vs 32, so if you prefer UR stay with 8-bits then we should refactor SYCL RT (and all adapters) to also do 8-bit and still **avoid conversions**.

Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/pi2ur.hpp:413]: @callumfare : can you update this PR and merge it? We shall look separately if we can avoid `fixupInfoValueTypes` by change bool to be 8-bits in PI.

**Review by smaslov-intel — APPROVED** [review]

**Comment by jandres742** [comment]
@intel/llvm-gatekeepers : please merge when possible.

**Pull request merged**